pkg/cmd/pr/merge/merge.go

```diff
@@ -228,14 +228,20 @@ func mergeRun(opts *MergeOptions) error {
 	}
 
 	if !isPRAlreadyMerged {
+
+		// ignore error, since git.Config will return "" for
+		// authorEmail if any error occurs
+		authorEmail, _ := git.Config("user.email")
```

Comment:
This respects the global setting, but also respects the per-checkout setting when inside of a git checkout of a repo. Should we respect the local setting even when merging a PR in another repository?

For example: I'm in the local directory of repoA, but I'm merging a PR in repoB: `gh pr merge --repo mislav/repoB 123`. Even though I've explicitly selected the repository to be another repository, `git config user.email` will read and apply the setting for repoA to my merge, ending in a potentially unwanted result.

Reply:
Ahh, yes that is an interesting case. Intuitively I believe the behaviour should be: if a repo is specified other than the current directory's repo, then always fall back to the GitHub default email.